app/routes/application.js

```diff
@@ -32,5 +35,16 @@ export default Ember.Route.extend(OSFAgnosticAuthRouteMixin, {
         if (locale) {
             this.set('i18n.locale', locale);
         }
+
+        // Check permissions
+        if (!this.get('session.isAuthenticated')) {
+            this.replaceWith('index');
+        } else {
+            this.get('currentUser.user').then((user) => {
```

Comment:
Ooh, I think *this* is when we want to return something from a model hook. The transition should pause until this promise resolves.

Reply:
👍  will try that